### PR TITLE
comptime error for mismatched types for some array functions

### PR DESCRIPTION
**Pull request by abhizer**
* update the docs to say that if the types don't match, the result is undefined
* fixes: #1475

Is this a user-visible change (yes/no): yes

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java [hunk 1]
```diff
@@ -71,21 +71,6 @@ public void testLeft() {
                 \s""");
     }
 
-    @Test
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java:74]: this test looks fine, why delete it?

**Review by mihaibudiu — APPROVED** [review]
you deleted a test too many

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java (hunk 1 above)
Comment (abhizer) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java:74]: it is already here: https://github.com/feldera/feldera/blob/89056b4b292cb1d433ce19e8eba60c8f4a6aa633/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayFunctionsTests.java#L11

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java [hunk 2]
```diff
@@ -1757,110 +1742,6 @@ public void testIssue1505() {
         );
     }
 
-    @Test
-    public void testArraysOverlap() {
-        this.qs("""
```
Comment (abhizer) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java:1762]: Moved to `ArrayFunctionsTests.java`.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java (hunk 1 above)
Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/functions/FunctionsTest.java:74]: the just go ahead and merge it when it passes the CI

**Pull request merged**